### ISSUE TITLE
UnderlineColour ignored

### ISSUE THREAD

**Issue by rovf** [mintty/mintty#673]
I have in my theme file

    UnderlineColour=153,241,219

and my TERM variable is set to xterm-256color, but when I output something which uses underlines, for instance display a man-page, the underlines are coloured as the ForegroundColour. 

mintty 2.7.6 (x86_64-pc-cygwin)

**Comment by mintty** [comment]
The line in your config file might have a CRLF lineend.
With 2.7.8, this would be stripped again.

**Comment by mintty** [comment]
Assuming my suggestion suits the issue, I'll close it.

**Comment by rovf** [comment]
Oops, I had sent you a comment on this issue, but for some reason this did not appear here!!!

There is no CR problem. All my files are ending in LF.

**Comment by mintty** [comment]
Please upload the config file in use. You may also try to modify it (e.g. different colour, colour name).

**Comment by rovf** [comment]
I solved the problem by updating mintty to version 2.7.6, which was made available a few days ago as a Cygwin package! So it seems that it had been fixed betwee 2.7.5 and 2.7.6.

**Comment by mintty** [comment]
If you updated to the current version, that would be 2.7.8, and would support my explanation; maybe you didn't recognize a CRLF lineend with your editor. Anyway, glad it's resolved.

**Comment by rovf** [comment]
Unfortunately, 2.7.6 is the most current one offered in the Cygwin package repository. Maybe I could manually compile the new version from the sources, but if this would be so easy in the Cygwin context, I guess the Cygwin maintainers would have done so already. Note that until very recently, the most recent version for Cygwin was even 2.7.5!

As for the line endings, I'm pretty sure that I did not have a CR. I analyzed the file using 

    `file FILENAME`

and this would classify a file with DOS line endings as such. In addition I did a 

    `od -cx FILENAME | grep 0d`

because the _file_ command looks only at the first couple of lines and might not detect correctly a file where some lines end in LF and some in CRLF, and here too, nothing was found.

My conclusion is that at least in the Cygwin version, the binary for 2.7.5 did not support this feature, but 2.7.6 does now. Could it be that this feature can be enabled or disabled by a compile time switch? In this case, it could be that the Cygwin maintainer did not enable this feature in the old version.

**Comment by mintty** [comment]
Hmm, it could be that this 2.7.6 change:
  * Manual underline adds less thickness in bold mode (#641).

had an impact. Also UnderlineColour is only applied if mintty does the underlining. Depending on font metrics, underlining can also be included in the font, thus being handled by Windows and not coloured then. That's why 2.7.1 had introduced this option:
  * New option UnderlineManual to enforce manual lining with default colour (#604).

We could analyze this further, but as the issue is resolved for you, I'd save the effort.

The latest cygwin version is actually 2.7.9 meanwhile. Maybe you refer to a download site that's a little bit behind...

And there are no features selectable by compile-time switches. Also, as cygwin is the native development environment for mintty, I'm actually the cygwin mintty maintainer myself :)

**Comment by rovf** [comment]
> The latest cygwin version is actually 2.7.9 meanwhile. Maybe you refer to a download site that's a little bit behind...

I thought they are all keeping in Sync :-O

I was always using http://ftp-stud.hs-esslingen.de/ . I will try other sites. Or could it be a difference between 32-Bit Cygwin and 64-Bit Cygwin? Can you recommend a site which is usually up-to-date?

**Comment by rovf** [comment]
Got one with 2.7.9 now!

**Comment by mintty** [comment]
The Esslingen site currently offers 2.7.9 to me, for both x-bit releases.

**Comment by rovf** [comment]
`mintty --version` still says 2.7.6, but perhaps this is because I have several mintty windows open, so the new version will be installed after the next reboot.

Thanks for caring....

**Comment by rovf** [comment]
BTW, I think part of the confusion also arose because mintty is displaying a wrong version string: After rebooting my system (to make sure that all pending installations are performed), I see the following:

    > mintty --version
    mintty 2.7.6 (x86_64-pc-cygwin)
    © 2013/2017 Andy Koppe / Thomas Wolff
    License GPLv3+: GNU GPL version 3 or later
    There is no warranty, to the extent permitted by law.

    -0-8- ~/tmp/coltest > cygcheck -c mintty
    Cygwin Package Information
    Package              Version        Status
    mintty               2.7.9-0        OK
    -0-8- ~/tmp/coltest >

You can see that the installed package is 2.7.9, but --version claims that it is 2.7.6

**Comment by mintty** [comment]
No, you can also check the version from the Options dialog, About... This should be consistent. If it's not the listed package version number, somehow the update has failed. Maybe you still had a terminal open during the update and the usual notification wasn't shown? Do you run a quiet update?

**Comment by rovf** [comment]
At least I rebooted after the update, but the option dialogue also says that I have 2.7.6. I don't do a quiet update, and if an application is open during update, I tell cygwin to go on and install the package after the reboot. 

I went now again into setup, but this time choosing "Reinstall" for the mintty package and *now* the version becomes 2.7.9, as it should. Something went wrong during the last install
